kernel/src/main/java/org/kframework/backend/kore/ModuleToKORE.java

```diff
@@ -106,6 +106,11 @@ public String convert() {
             Att att = module.sortAttributesFor().get(sort).getOrElse(() -> KORE.Att());
             if (att.contains(Attribute.HOOK_KEY) && !dummyHookedSorts.contains(att.get(Attribute.HOOK_KEY))) {
                 sb.append("hooked-");
+            } else if (att.contains(Attribute.HOOK_KEY)) {
```

Comment:
I find the logic here a bit hard to follow.  Do you mean that it contains the `HOOK_KEY`, but it is in the `dummyHookedSorts`?  If so, wouldn't it be better to merge the two `if`s and have one only for `!dummyHookedSorts`?